tests/ci/build_report_check.py

```diff
@@ -88,8 +102,9 @@ def main():
     # The code to collect missing reports for failed jobs
     missing_job_names = [
         name
-        for name in needs_data
+        for name, data in needs_data.items()
         if not any(1 for br in build_results if br.job_name.startswith(name))
+        and data["result"] != "skipped"
```

Comment:
i'm afraid these changes could be dangerous for the public repo. 
It looks like we trade off reliability of whole ci for a tiny use case

Reply:
why? What would be the example?

We always run the builds now, right?

Reply:
This tiny thing prevents us from automatically merging sync PRs.

Reply:
For the reference, if every build is skipped, we'll still have it catched https://github.com/ClickHouse/ClickHouse/blob/6576eece720652a7ba16f053a2f91c7f8d6dd1d5/tests/ci/build_report_check.py#L153-L158

Reply:
no we don't run builds always

Reply:
`build check report` now provides means to check that all builds are ok.

Reply:
Ok, but let's find the example that would break.

Because the issue is still pretty major, see every linked issue and PR

Reply:
We can use new logic only for specific branches (for example, if the branch name starts with `"sync-upstream/"`)

Reply:
we discussed it by voice

For now, there's no way it would break. In the future, after some workflow yamls refactoring, it may be a general problem. But it will be refactoring in any case, so now it's safe enough to keep it as is